### ISSUE TITLE
Pull request workflow - require up-to-date branches?

### ISSUE THREAD

**Issue by psalz** [KhronosGroup/SYCL-CTS#233]
We currently have a problem where a PR (#203) that predates the CI setup was merged, breaking compilation for ComputeCpp. This then lead to all new PRs in the last month to also fail CI. While CI has been fixed, those existing PRs are still marked as failed, and re-running the GitHub actions unfortunately doesn't help: While the `pull_request` trigger used for our workflow first merges the source branch into the base (target) branch and then runs the tests, it seems that re-running does *not* merge into the current version of the base branch, instead reusing the revision from when the PR was created (a bit of a limitation on GitHub's part, in my opinion).

There are basically two ways how the existing PRs can be run against the current `SYCL-2020` branch:
1. The author force-pushes the source branch (at least I think that will then use the current base branch, haven't tested it though)
2. The base branch is back-merged into the source branch

To avoid further breakage from these now untested PRs I've temporarily enabled branch protection on the `SYCL-2020` branch, which requires that all CI checks pass before we can merge, and also requires that the source branch is *up-to-date*. The latter setting causes GitHub to show the following button:

![image](https://user-images.githubusercontent.com/791348/145368226-1732fa87-c089-4a5a-8f9f-bedd371a217d.png)

I've tried this for #230, where you can see that while DPCPP is still running at the time of writing, the ComputeCpp run has already passed.

Now my question is, should we keep this setting in general? The advantage is that we're guaranteed that merging a PR won't break CI. The downside is that we'll introduce more noise into the history with all those back-merge commits.

**Comment by psalz** [comment]
WG Meeting: Only do this if merges breaking CI becomes a frequent issue.
